dndsim/src/classes/Barbarian.ts

```diff
@@ -0,0 +1,211 @@
+            event.damage.flatDmg += this.damageBonus
+        }
+    }
+
+    shortRest(character: Character) {
+        character.removeEffect(RageEffect)
+    }
+
+    addResource(character: Character) {
+        character.resources.set(
+            RageResource,
+            new Resource({
+                name: RageResource,
+                character,
+                initialMax: 2,
+                incrementOnShortRest: true,
+                resetOnLongRest: true,
+
+            })
+        )
+    }
+}
+
+class AddRage extends Feat {
+    apply(character: Character): void {
+        character.getResource(RageResource).addMax(1)
+    }
+}
+
+class AddRageDamage extends Feat {
+    apply(character: Character): void {
+        const rageFeat = character.feats.find((feat) => feat instanceof Rage) as Rage | undefined
+        if (!rageFeat) {
+            return
+        }
+        rageFeat.damageBonus += 1
+    }
+}
+
+class RecklessAttack extends Feat {
+    apply(character: Character) {
+        character.events.on("attack_roll", (event) => this.attackRoll(character, event))
+    }
+
+    attackRoll(character: Character, event: AttackRollEvent) {
+        if (event.attack?.attack?.weapon()?.mod(character) == "str") {
+            event.adv = true
+            event.addTag(RecklessTag)
+        }
+    }
+}
+
+class PrimalChampion extends Feat {
+    apply(character: Character) {
+        character.increaseStatAndMax("str", 4)
+        character.increaseStatAndMax("con", 4)
+    }
+}
+
+class Frenzy extends Feat {
+    applied = false
+
+    apply(character: Character) {
+        character.events.on("begin_turn", () => this.applied = false)
+        character.events.on("damage_roll", (event) => this.damageRoll(character, event))
+    }
+
+    damageRoll(character: Character, event: DamageRollEvent) {
+        if (!this.applied && event.attack?.hasTag(RecklessTag)) {
+            const rageFeat = character.feats.find((feat) => feat instanceof Rage) as Rage | undefined
+            if (!rageFeat) {
+                return
+            }
+            const dice = Array(rageFeat.damageBonus).fill(6)
+            event.damage.addDice(dice)
+            this.applied = true
+        }
+    }
+}
+
+export class Barbarian {
+    static baseFeats(args: {
+        level: number
+        asis: Array<Feat>
+        masteries: WeaponMastery[]
+    }): Feat[] {
+        const { level, asis, masteries } = args
+        const feats: Feat[] = []
+        if (level >= 1) {
+            feats.push(new ClassLevel("Barbarian", level))
+            feats.push(new WeaponMasteries(masteries))
+            feats.push(new Rage(2))
+        }
+        if (level >= 2) {
+            feats.push(new RecklessAttack())
+        }
+        if (level >= 5) {
+            feats.push(new ExtraAttack(2))
+        }
+        if (level >= 6) {
+            feats.push(new AddRage())
+        }
+        if (level >= 9) {
```

Comment:
I didn't add Brutal Strike since it's difficult to decide whether to forgo advantage for the extra 1d10, then later 2d10 damage.